osu.Game/Database/RealmAccess.cs

```diff
@@ -649,8 +647,14 @@ void restoreOperation()
             {
                 Logger.Log(@"Restoring realm operations.", LoggingTarget.Database);
                 realmRetrievalLock.Release();
+
                 // Post back to the update thread to revive any subscriptions.
-                syncContext?.Post(_ => ensureUpdateRealm(), null);
+                // In the case we are on the update thread, let's also require this to run synchronously.
+                // This requirement is mostly due to test coverage, but shouldn't cause any harm.
+                if (ThreadSafety.IsUpdateThread)
+                    syncContext?.Send(_ => ensureUpdateRealm(), null);
```

Comment:
So I guess these `Send()` calls and their inner "flush the scheduler" calls are supposed to enforce that the order of execution of these realm callbacks is consistent with the order in which they were posted/sent - but in that case, should the following also use `Send()` in the update thread branch?

https://github.com/ppy/osu/blob/4a9f4eecba45e66428de23094f1d1387199bb450/osu.Game/Database/RealmAccess.cs#L298-L301

Because that looks like the mirror image of this situation.

Reply:
It shouldn't matter here as well, both will provide non-incorrect behaviour. which do you think reads better?

to rephrase, both could be direct calls of that is preferred.

Reply:
I actually think I prefer using `.Send()` in both cases, having the guarantee of in-enqueue-order execution seems valuable. But that's not a super strong opinion, I'm mostly looking for consistency.